incomplete_cooperative/generators.py

```diff
@@ -197,6 +197,41 @@ def xos(number_of_players: int, generator: np.random.Generator = np.random.defau
     return ig
 
 
+def xs(number_of_players: int, generator: np.random.Generator = np.random.default_rng()) -> IncompleteCooperativeGame:
+    """Generate a random XS function."""
+    singletons = np.array([generator.random() for _ in range(number_of_players)])
+    coalitions = all_coalitions(number_of_players)
+    players_in_coalitions = [list(coalition.players) for coalition in coalitions]
+    values = np.array([np.max(singletons[players], initial=0) for players in players_in_coalitions])
```

Comment:
Do we this this is one of the computational bottlenecks? I would think there is a way to do all this in numpy.

Reply:
This is all one run through an iterator. We'd need to do this in Python either way. I don't know how to transfer more of this into numpy.

Also, it's relatively fast compared to the stiluff we do in the OR operation